### PR TITLE
Do not obbey rel=self if WebSub is disabled

### PR DESCRIPTION
**Pull request by Alkarex**
https://github.com/FreshRSS/FreshRSS/issues/2654
https://github.com/pubsubhubbub/PubSubHubbub/wiki/Moving-Feeds-or-changing-Hubs

### CODE REVIEW

**Review by Frenzie — COMMENTED** [review]
Not sure what I think of the change but the code is fine of course.

**Review by marienfressinaud — COMMENTED** [review]
I'm not totally comfortable with this since it basically provides a workaround to a problem which is not about WebSub but about "why FRSS changes automatically feed urls"… We can discuss about the interest of this change, but I don't think it satisfies the initial issue

**Comment by Frenzie** [comment]
I agree, I think that a fix to the (supposed?) issue would be some kind of `Prevent automatic URL update` checkbox or dropdown, similar to a setting like `Always clear cache` or `Verify SSL security`.

Edit: It's probably better to phrase it positively rather than negatively, along the lines of `Apply/Follow/Obey URL updates`.

**Comment by Alkarex** [comment]
This patch only indirectly addresses the cases of some automatic URL redirections. Indeed, the `self` part is an element of the WebSub protocol. As such, when disabling WebSub, it makes sense not to process it.

**Comment by marienfressinaud** [comment]
Indeed, it makes sense, but shouldn't we use `$feed->pubSubHubbubEnabled()` instead?

File: app/Controllers/feedController.php [hunk 1]
```diff
@@ -415,7 +415,7 @@ public static function actualizeFeed($feed_id, $feed_url, $force, $simplePiePush
 				$entryDAO->commit();
 			}
 
-			if ($feed->hubUrl() && $feed->selfUrl()) {	//selfUrl has priority for WebSub
+			if ($pubsubhubbubEnabledGeneral && $feed->hubUrl() && $feed->selfUrl()) {	//selfUrl has priority for WebSub
```
Comment (marienfressinaud) [inline app/Controllers/feedController.php]:
```suggestion
			if ($feed->pubSubHubbubEnabled() && $feed->hubUrl() && $feed->selfUrl()) {	//selfUrl has priority for WebSub
```

I think `$feed->pubSubHubbubEnabled()` returns `false` when `$pubsubhubbubEnabledGeneral` is `false` and I find it's more logical this way

Comment (Alkarex) [reply app/Controllers/feedController.php]: `$feed->pubSubHubbubEnabled()` would not work, but you are right that it was not the correct variable :-)
Fix coming using 
https://github.com/FreshRSS/FreshRSS/blob/1ddf98c50b2e0388c76386202bd8be3442640418/app/Controllers/feedController.php#L279

**Comment by Alkarex** [comment]
Any final comment? :-)

File: app/Controllers/feedController.php [hunk 2]
```diff
@@ -415,7 +415,7 @@ public static function actualizeFeed($feed_id, $feed_url, $force, $simplePiePush
 				$entryDAO->commit();
 			}
 
-			if ($feed->hubUrl() && $feed->selfUrl()) {	//selfUrl has priority for WebSub
+			if ($pubSubHubbubEnabled && $feed->hubUrl() && $feed->selfUrl()) {	//selfUrl has priority for WebSub
 				if ($feed->selfUrl() !== $url) {	//https://code.google.com/p/pubsubhubbub/wiki/MovingFeedsOrChangingHubs
```
Comment (Frenzie) [inline app/Controllers/feedController.php]: This URL no longer works.

Comment (Alkarex) [reply app/Controllers/feedController.php]:
```suggestion
				if ($feed->selfUrl() !== $url) {	// https://github.com/pubsubhubbub/PubSubHubbub/wiki/Moving-Feeds-or-changing-Hubs
```

Comment (Alkarex) [reply app/Controllers/feedController.php]: Fixed :-)

**Review by marienfressinaud — APPROVED** [review]

File: app/Controllers/feedController.php (hunk 1 above)
Comment (Alkarex) [reply app/Controllers/feedController.php]: Reverted to the first version https://github.com/FreshRSS/FreshRSS/pull/2922

**Pull request merged**